cmd/batch-handlers.go

```diff
@@ -1212,12 +1212,39 @@ func batchReplicationTrace(d batchReplicationMetric, job string, startTime time.
 	}
 }
 
-func (m *batchJobMetrics) report(jobID string) (metrics *madmin.BatchJobMetrics) {
-	metrics = &madmin.BatchJobMetrics{CollectedAt: time.Now(), Jobs: make(map[string]madmin.JobMetric)}
+func (m *batchJobMetrics) report(jobID string) *madmin.BatchJobMetrics {
 	m.RLock()
 	defer m.RUnlock()
+	if jobID != "" {
+		job, ok := m.metrics[jobID]
+		if !ok {
+			return nil
+		}
+		return &madmin.BatchJobMetrics{
+			CollectedAt: time.Now(),
+			Jobs: map[string]madmin.JobMetric{
+				jobID: {
+					JobID:         job.JobID,
+					JobType:       job.JobType,
+					StartTime:     job.StartTime,
+					LastUpdate:    job.LastUpdate,
+					RetryAttempts: job.RetryAttempts,
+					Complete:      job.Complete,
+					Failed:        job.Failed,
+					Replicate: &madmin.ReplicateInfo{
+						Bucket:           job.Bucket,
+						Object:           job.Object,
+						Objects:          job.Objects,
+						ObjectsFailed:    job.ObjectsFailed,
+						BytesTransferred: job.BytesTransferred,
+						BytesFailed:      job.BytesFailed,
+					},
+				},
+			},
+		}
+	}
```

Comment:
the reason it was intentionally not written this way is because of the amount of duplicate code here. 

that's why traversing `m.metrics` was okay because that is a cheap operation.